infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -102,20 +90,22 @@ private boolean processMsg(LogReplicationEntryMsg txMessage) {
 
             for (OpaqueEntry opaqueEntry : newOpaqueEntryList) {
                 for (UUID streamId : opaqueEntry.getEntries().keySet()) {
-                    if (!streamMap.containsKey(streamId)) {
-                        log.warn("Skip applying log entries for stream {} as it is noisy. LR could be undergoing a rolling upgrade", streamId);
-                        continue;
-                    }
-
                     List<SMREntry> smrEntries = opaqueEntry.getEntries().get(streamId);
                     if (streamId.equals(REGISTRY_TABLE_ID)) {
-                        // Only retain tables that are not present in the registry table
-                        smrEntries = fetchNewEntries(new ArrayList<>(smrEntries));
-                    }
-
-                    for (SMREntry smrEntry : smrEntries) {
-                        // If stream tags exist for the current stream, it means its intended for streaming on the Sink (receiver)
-                        txnContext.logUpdate(streamId, smrEntry, dataStreamToTagsMap.get(streamId));
+                        // Preprocess registry table entries to add their serialization info.
+                        smrEntries = addSerializationInfo(smrEntries);
+                        log.info("{} entries for registry table found during log entry sync.", smrEntries.size());
+                        for (SMREntry smrEntry : smrEntries) {
+                            // If stream tags exist for the current stream, it means it is intended for streaming on the Sink (receiver)
+                            txnContext.logUpdate(streamId, smrEntry, config.getDataStreamToTagsMap().get(streamId));
+                        }
+                        // Update the config after handling registry table entries
+                        config.syncWithRegistry();
```

Comment:
this sync will not get the updates to the registry table made in L100 because the transaction is not yet committed.  You will have to pass the same txn to read these writes and add them to the config.

It will get addressed in PR 3331 which will preserve the transaction boundaries.